include/libtorrent/aux_/tracker_manager.hpp

```diff
@@ -392,6 +401,9 @@ using tracker_request_flags_t = flags::bitfield_flag<std::uint8_t, struct tracke
 		aux::resolver_interface& m_host_resolver;
 		aux::session_settings const& m_settings;
 		counters& m_stats_counters;
+#ifdef TORRENT_USE_LIBCURL
+		std::shared_ptr<aux::curl_thread_manager> m_curl_thread_manager;
```

Comment:
is it  really necessary for this to be held by a `shared_ptr`? Why can't this class be the sole owner?